rasa/utils/tensorflow/models.py

```diff
@@ -361,14 +362,16 @@ def _merge_batch_outputs(
     def _empty_lists_to_none_in_dict(input_dict: Dict[Text, Any]) -> Dict[Text, Any]:
         """Recursively replaces empty list or np array with None in a dictionary."""
 
-        def _recurse(x: Union[Dict[Text, Any], List[Any], np.ndarray]) -> Optional[Any]:
+        def _recurse(
+            x: Union[Dict[Text, Any], List[Any], np.ndarray]
+        ) -> Optional[Union[Dict[Text, Any], List[np.ndarray]]]:
             if isinstance(x, dict):
                 return {k: _recurse(v) for k, v in x.items()}
             elif (isinstance(x, list) or isinstance(x, np.ndarray)) and np.size(x) == 0:
                 return None
             return x
 
-        return _recurse(input_dict)
+        return {k: _recurse(v) for k, v in input_dict.items()}
```

Comment:
Why change this? The base condition on line 368 already takes care of it isn't it?

Reply:
That's one of those changes that comes from the merge with main.

Reply:
It'll just save one isinstance check. I'll just leave it in, doesn't hurt.